message_queue.cpp

```diff
@@ -0,0 +1,32 @@
+#include <iostream>
+#include <queue>
+#include <mutex>
+
+using namespace std;
+
+class MessageQueue {
```

Comment:
Add comment here saying what this class does.
```
/**
 * This class implements ...
 */
```
Also add comments for each member function below.